src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php

```diff
@@ -85,15 +85,20 @@ protected function removePrices(ProductInterface $product, AttributeInterface $a
     {
         $productValue = $product->getValue($attribute->getCode(), $locale, $scope);
 
+        $currencyToRemove = [];
+        foreach ($data as $priceToRemove) {
+            $currencyToRemove[] = $priceToRemove['currency'];
+        }
+
         if (null !== $productValue) {
             $prices = [];
             foreach ($productValue->getData() as $price) {
-                if (!in_array(['amount' => $price->getData(), 'currency' => $price->getCurrency()], $data)) {
```

Comment:
In fact, this is how 1.7 is working. It does not check the amount but removes the price for the given currency: https://github.com/akeneo/pim-community-dev/blob/ba189044dd93fb14042de97ba6a0a2ce8382e642/src/Pim/Component/Catalog/Updater/Remover/PriceCollectionAttributeRemover.php#L84-L84